wallet/src/wallet/mod.rs

```diff
@@ -613,12 +617,16 @@ impl<B: storage::Backend> Wallet<B> {
     }
 
     pub fn get_pos_gen_block_data(
-        &mut self,
+        &self,
         account_index: U31,
+        pool_id: PoolId,
     ) -> WalletResult<PoSGenerateBlockInputData> {
         let db_tx = self.db.transaction_ro_unlocked()?;
-        self.get_account(account_index)?
-            .get_pos_gen_block_data(&db_tx, self.latest_median_time)
+        self.get_account(account_index)?.get_pos_gen_block_data(
+            &db_tx,
+            self.latest_median_time,
```

Comment:
If multiple pools are staking within the same wallet, dosen't that mean that each can have a different latest median time?

Reply:
The median time is always stored from the node (without looking at the wallet state), so it's theoretically possible. But the staking doesn't start until the wallet is fully synchronized, so this shouldn't be a problem.